### ISSUE TITLE
Add read_concurrency_semaphore metrics

### ISSUE THREAD

**Issue by vladzcloudius** [scylladb/scylladb#20993]
HEAD: 8bfbc563cc6126cc184e6f64d4c590d54ec600a8

## Description

In order to be able issues related to read_concurrency_semaphore functioning we should add/change following metrics:

Change to existing metrics:
* active (`scylla_database_active_reads`) and queued (`scylla_database_queued_reads`) requests should be changed from being a "gauge" to "derived" in order not to miss spikes

Add new metrics:
* Time tasks spent in r_c_s queue
* Amount of memory used so far
* Number of tasks waiting for CPU limit
* Number of tasks waiting for I/O limits 
* Expose the rest of metrics printed during the r_c_s timeout dump

**Comment by vladzcloudius** [comment]
cc @denesb

**Comment by denesb** [comment]
@amnonh are we allowed to change the type of existing metrics? Maybe if we coordinate with the monitoring stack?

**Comment by amnonh** [comment]
We can change the metrics type if needed (it's not derived. It's counter, but just as well), but it needs to be done on a minor version (as opposed to a patch version).

When adding new metrics, you should consider how helpful it will be and the overall effect in a large cluster.

**Comment by denesb** [comment]
@vladzcloudius I don't think we can just expose all those stats as metrics. We need to think about what are the most high-value metrics and only expose those. We already have a problem with too many metrics.

**Comment by vladzcloudius** [comment]
> [@vladzcloudius](https://github.com/vladzcloudius) I don't think we can just expose all those stats as metrics. We need to think about what are the most high-value metrics and only expose those. We already have a problem with too many metrics.

I'm not familiar with the "too many metrics" problem, especially after we stopped sending always-zero metrics.
But even before that I remember us only having issues with histograms. Am I wrong, @amnonh ?

Regardless, @denesb, let's start with first 4 bullets above + "amount of all tasks waiting in the r_c_s queue"

**Comment by vladzcloudius** [comment]
Another two metrics we may want are
* Number of admitted tasks ("counter")
* Number of tasks that released the semaphore ("counter")

This would allow us to also see the amount of tasks that were admitted but not yet released the semaphore:
`rate(num_admitted_tasks) - rate(num_sem_release)`
